tools/chorctl/internal/common/common.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2025 STRATO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	pb "github.com/clyso/chorus/proto/gen/go/chorus"
+)
+
+type ReplNameBuilder func(*pb.Replication) string
+
+func NewReplNameBuilder(inFormat *string) (ReplNameBuilder, error) {
+	isEscaped := false
+	outFormat := ""
+	outFuncs := []ReplNameBuilder{}
+	for i, c := range *inFormat {
```

Comment:
should we check if `inFormat` is nil to avoid panic?

maybe it will be easier to just pass it as a string.